ARCs/arc-0084.md

```diff
@@ -0,0 +1,750 @@
+          "type": "((address,address,uint64),(uint64,uint64,uint64,uint64,uint64))[]"
+        }
+      ],
+      "returns": {
+        "type": "void"
+      },
+      "actions": {
+        "create": [],
+        "call": [
+          "NoOp"
+        ]
+      }
+    },
+    {
+      "name": "doTransfers",
+      "desc": "*********************\nTransfer/Mint Methods\n*********************",
+      "args": [
+        {
+          "name": "sender",
+          "type": "address"
+        },
+        {
+          "name": "transfers",
+          "type": "(uint64,address,address,uint64)[]"
+        }
+      ],
+      "returns": {
+        "type": "void"
+      },
+      "actions": {
+        "create": [],
+        "call": [
+          "NoOp"
+        ]
+      }
+    },
+    {
+      "name": "arc84_mint",
+      "args": [
+        {
+          "name": "collectionId",
+          "type": "uint64"
+        },
+        {
+          "name": "params",
+          "type": "(byte[32],byte[8],uint64,uint64,address,uint64)",
+          "struct": "Params"
+        }
+      ],
+      "returns": {
+        "type": "uint64"
+      },
+      "actions": {
+        "create": [],
+        "call": [
+          "NoOp"
+        ]
+      }
+    }
+  ]
+}
+```
+
+</details>
+
+#### TEALScript Data Interface Signatures
+
+```typescript
+
+export type Params = {
+  /** The descriptive name of the token */
+  name: bytes<32>;
+  /** The short ticker symbol of the token */
+  symbol: bytes<8>;
+  /** The total supply */
+  total: uint64;
+  /** The number of digits to use after the decimal point when displaying the asset */
+  decimals: uint64;
+  /** The address that can modify the token metadata */
+  manager: Address;
+  /** The address of the app that implements the transfer hook. If this is not set, transfers are always approved */
+  transferHookApp: AppID;
+};
+
+export type IdAndAddress = {
+  tokenId: TokenId;
+  address: Address;
+};
+
+export type MetadataKey = {
+  /** id for collection or token depending on context */
+  id: uint64;
+  key: bytes;
+};
+
+export type Metadata = {
+  mutable: boolean;
+  data: bytes;
+};
+
+export type AllowanceKey = {
+  holder: Address;
+  sender: Address;
+  tokenId: TokenId;
+};
+
+export type Allowance = {
+  amount: uint64;
+  remainingAmount: uint64;
+  cooldown: uint64;
+  lastUsed: uint64;
+  expirationTimestamp: uint64;
+};
+
+export type Collection = {
+  /** The total number of tokens minted in this collection */
+  minted: uint64;
+
+  /** The cap to total tokens that can be minted */
+  mintCap: uint64;
+
+  /** The address that can modify collection metadata and mint new tokens */
+  manager: Address;
+};
+
+arc84_newCollection(manager: Address, mintCap: uint64): CollectionId 
+
+/** ***************
+* Getter Methods
+**************** */
+
+arc84_collection_minted(id: CollectionId): uint64 
+
+arc84_metadata(key: MetadataKey): Metadata 
+
+arc84_balanceOf(id: TokenId, account: Address): uint64 
```

Comment:
I'd also recommend an arc84_numHoldersOf(id: TokenId).
Determining the number of holders for a token is a common need, and with a global data registry, fetching literally all boxes (or even prefix-match fetch of all) will be painful, yet updating a counter on transitions from/to 0 balance will be easy to handle.